enumerable_methods.rb

```diff
@@ -0,0 +1,90 @@
+# Project 2: Enumerable Methods
+module Enumerable
+  def my_each(&block)
+    size.times { |i| block.call(self[i]) } and return self if block
+    to_enum
+  end
+
+  def my_each_with_index(&block)
+    size.times { |idx| block.call(self[idx], idx) } and return self if block_given?
+    to_enum
+  end
+
+  def my_select(&block)
+    selection = []
+    my_each { |item| selection << item if block.call(item) } and return selection if block_given?
+    to_enum
+  end
+
+  def my_map
+    mapped = []
+    my_each { |item| mapped << yield(item) if yield(item) != 0 } and return mapped if block_given?
+    to_enum
+  end
+
+  def my_all?(args = nil, &block)
+    swap = true
+    if block
+      my_each { |item| swap = false unless block.call(item) }
+    elsif args.nil?
+      my_each { |item| swap = false unless item }
+    else
+      my_each { |item| swap = false unless args === item }
+      to_enum
+    end
+    swap
+  end
+
+  def my_any?(args = nil, &block)
+    swap = false
+    if block
+      my_each { |item| swap = true if block.call(item) }
+    elsif args.nil?
+      my_each { |item| swap = true if item }
+    else
+      my_each { |item| swap = true if args === item }
+      to_enum
+    end
+    swap
+  end
+
+  def my_none?(args = nil, &block)
+    swap = true
+    if block
+      my_each { |item| swap = false if block.call(item) }
+    elsif args.nil?
+      my_each { |item| swap = false if item }
+    else
+      my_each { |item| swap = false if args === item }
+      to_enum
+    end
+    swap
+  end
+
+  def my_count(args = nilm, &block)
+    count = 0
+    if block_given?
+      my_each { |item| count += 1 if block.call(item) }
+    elsif args.nil?
+      my_each { |item| count += 1 if item }
+    else
+      my_each { |item| count += 1 if args === item }
+      to_enum
+    end
+    count
+  end
```

Comment:
* [ ] my_count returns the number of items in enum through enumeration
  ```ruby
    array = [1,2,5,4,7]
    p array.my_count == array.count # returns this error
      # NameError:
       undefined local variable or method `nilm' for #<Array:0x000056468bea49a8>
       Did you mean?  nil
                      nil?
     # ./enumerable_methods.rb:64:in `my_count'
  ```

* [ ] my_count counts the number of elements yielding a true value if a block is given
```ruby
array = [1,2,8,9,7,5,6,3,11]
block = proc { |num| num < 5 }
array.my_count(&block) == array.count(&block) # returns this error
   # NameError:
       undefined local variable or method `nilm' for #<Array:0x000056468bec32e0>
       Did you mean?  nil
                      nil?
     # ./enumerable_methods.rb:64:in `my_count'
```